### ISSUE TITLE
VMware 4.11.2.0 system VMs memory consumption grows overtime until heavy swapping occurs

### ISSUE THREAD

**Issue by rhtyd** [apache/cloudstack#3039]
Memory Consumption on vSphere
vRouter are starting to swap with low memory available.
Increasing memory only causes a longer time period before heavy swapping occurs.

@resmo shared a related issue on dev list: https://github.com/systemd/systemd/issues/8015

##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
VR
~~~

##### CLOUDSTACK VERSION

~~~
ACS 4.11.x
~~~

##### OS / ENVIRONMENT

Seen only in VMware environments

##### STEPS TO REPRODUCE

~~~
- Deploy VR in vmware
- SSH into it from management server
- See memory growth over time
~~~

##### EXPECTED RESULTS

~~~
No additional memory consumption.
~~~

**Comment by onitake** [comment]
Hi @rhtyd @PaulAngus - thanks for taking on this issue.

If you have any suggestions on more tests we could run, please share.

**Comment by onitake** [comment]
Is the fix going to be something like the one mentioned here https://lists.apache.org/thread.html/cb36b5efe8327c080919086688707650a3ff036c884a2dd87dd6e66d@%3Cdev.cloudstack.apache.org%3E , or is this an unrelated issue?

**Comment by onitake** [comment]
We were able to drive memory usage through the roof with SSH connections, because each session would leak memory. But we also had success with lots of network reconfiguration requests, so I guess both are valid test cases.

**Comment by onitake** [comment]
@PaulAngus Is it also possible to get just the templates in lieu of full 4.11.2.1 RPMs?
If I try to access your package server through a web browser, I just get a blank page.

**Comment by andrijapanicsb** [comment]
Any progress on this one - anyone else to share experience or a proper fix ?

**Comment by PaulAngus** [comment]
This is addressed by https://github.com/apache/cloudstack/pull/3127

**Comment by andrijapanicsb** [comment]
Pardon my blind eyes ... ... ...

**Comment by rhtyd** [comment]
Interesting notes from https://github.com/systemd/systemd/issues/8015#issuecomment-475324192
```
On Debian I fix the memory consumption commenting the following line on /etc/pam.d/common-session:

#session optional pam_systemd.so

But, with that, the ssh sessions doesn't terminate correctly. The fix is already done:

cp /usr/share/doc/openssh-client/examples/ssh-session-cleanup.service /etc/systemd/system/
systemctl  enable ssh-session-cleanup.service
Other alternative solution for Debian, add to the end of the file /etc/pam.d/systemd-user :

@include null

It doesn't include any file, because null doesn't exists.
```

**Comment by andrijapanicsb** [comment]
^^^ included in #3126 and #3127

**Comment by rhtyd** [comment]
Carry on here: https://github.com/apache/cloudstack/pull/3126
